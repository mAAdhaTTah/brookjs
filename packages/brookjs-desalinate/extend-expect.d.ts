import 'jest-kefir/extend-expect';

declare namespace jest {
  interface Matchers<R, T> {
    toEmitFromDelta<V, E>(
      expect: import('kefir-test-utils').EventWithTime<V, E>[],
      cb?: (
        send: (state: any, action: any) => void,
        tick: (s: number) => void,
        clock: import('lolex').Clock
      ) => void,
      opts?: { timeLimit?: number }
    ): R;
    toEmitFromJunction<V, E>(
      expect: import('kefir-test-utils').EventWithTime<V, E>[],
      cb?: (
        api: import('@testing-library/react').RenderResult,
        tick: (s: number) => void,
        clock: import('lolex').Clock
      ) => void,
      opts?: { timeLimit?: number }
    ): R;
  }
}
