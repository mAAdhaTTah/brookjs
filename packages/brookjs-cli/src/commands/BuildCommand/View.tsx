import webpack from 'webpack';
import Spinner from 'ink-spinner';
import { Box, Color, useApp } from 'ink';
import React, { useEffect } from 'react';
import { format } from 'date-fns';
import { ExitError, useExit } from '../../cli';
import { State } from './types';

const Asset: React.FC<{
  name: string;
  entry: webpack.Stats.ChunkGroup;
}> = ({ name }) => {
  return (
    <Box flexDirection="row" paddingLeft={2}>
      <Color magentaBright>
        {/* eslint-disable-next-line jsx-a11y/accessible-emoji */}
        <Box marginRight={2}>ðŸ¤–</Box>
        {name}
      </Color>
    </Box>
  );
};

const Warning: React.FC<{ message: string }> = ({ message }) => {
  return (
    <Box>
      <Color yellow>{message}</Color>
    </Box>
  );
};

const ErrorMsg: React.FC<{ message: string }> = ({ message }) => {
  return (
    <Box>
      <Color redBright>{message}</Color>
    </Box>
  );
};

const Building: React.FC<{}> = () => (
  <Box>
    <Color green>
      <Spinner type="arrow3" />
    </Color>
    <Box paddingLeft={1}>Building application...</Box>
  </Box>
);

const Built: React.FC<{ results: webpack.Stats; watch: boolean }> = ({
  results,
  watch
}) => {
  const {
    assets = [],
    entrypoints = {},
    builtAt = Date.now(),
    warnings,
    errors
  } = results.toJson('verbose');

  return (
    <Box flexDirection="column">
      <Color cyan>{format(builtAt, "'Finished compiling at 'HH:mm:ss")}</Color>
      <Color cyanBright>
        Generated {assets.length} asset{assets.length === 1 ? '' : 's'} from
        entrypoints:
      </Color>
      <Box flexDirection="column" marginBottom={1}>
        {Object.entries(entrypoints).map(([name, entry]) => (
          <Asset key={name} name={name} entry={entry} />
        ))}
      </Box>
      {warnings.length > 0 && (
        <Box flexDirection="column" marginBottom={1}>
          <Box marginBottom={1}>
            <Color yellowBright>Compilation warnings:</Color>
          </Box>
          {warnings.map((warning, i) => (
            <Warning key={i} message={warning} />
          ))}
        </Box>
      )}
      {errors.length > 0 && <BuildErrors errors={errors} watch={watch} />}
    </Box>
  );
};

const BuildErrors: React.FC<{ errors: string[]; watch: boolean }> = ({
  errors,
  watch
}) => {
  const { exit } = useApp();

  useEffect(() => {
    if (!watch) {
      exit(new ExitError(1));
    }
  }, [watch, exit]);

  return (
    <Box flexDirection="column" marginBottom={1}>
      <Box marginBottom={1}>
        <Color redBright>Compilation errors:</Color>
      </Box>
      {errors.map((error, i) => (
        <ErrorMsg key={i} message={error} />
      ))}
    </Box>
  );
};

const RCNotLoaded: React.FC = () => {
  useExit(new ExitError(1));

  return <Color red>RC file not loaded. Cannot build.</Color>;
};

const RCInvalid: React.FC<{ message: string }> = ({ message }) => {
  useExit(new ExitError(1));

  return <Color red>RC file invalid. {message}</Color>;
};

const View: React.FC<State> = props => {
  if (props.rc == null) {
    return <RCNotLoaded />;
  }

  if (props.rc instanceof Error) {
    return <RCInvalid message={props.rc.message} />;
  }

  if (props.building) {
    return <Building />;
  }

  if (props.results instanceof Error) {
    return <BuildErrors watch={props.watch} errors={[props.results.message]} />;
  }

  if (!props.watch && props.results.hasErrors()) {
    const { errors } = props.results.toJson('errors-only');

    return <BuildErrors watch={props.watch} errors={errors} />;
  }

  return <Built results={props.results} watch={props.watch} />;
};

export default View;
