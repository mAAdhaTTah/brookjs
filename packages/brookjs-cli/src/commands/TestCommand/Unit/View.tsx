import { Box, Color } from 'ink';
import React from 'react';
import { useExit } from '../../../cli';
import { State, CompleteState, ErrorState } from './types';

const Running: React.FC = React.memo(() => (
  <Box>
    <Color yellow>Running Jest...</Color>
  </Box>
));

const Results: React.FC = () => {
  useExit();

  return (
    <Box>
      <Color green>Finished running tests</Color>
    </Box>
  );
};

const isDone = (state: State): state is CompleteState | ErrorState =>
  state.status === 'complete';

const View: React.FC<State> = props => {
  if (props.rc == null) {
    return <Color red>RC file not loaded. Cannot build.</Color>;
  }

  if (props.rc instanceof Error) {
    return <Color red>RC file invalid. {props.rc.message}</Color>;
  }

  if (!isDone(props)) {
    return <Running />;
  }

  return <Results />;
};

export default View;
