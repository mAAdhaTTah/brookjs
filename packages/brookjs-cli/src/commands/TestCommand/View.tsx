import Spinner from 'ink-spinner';
import { Box, Color } from 'ink';
import React from 'react';
import { Nullable } from 'typescript-nullable';
import { useExit } from '../../cli';
import { State, CompleteState } from './types';

const Running: React.FC<{ command: string }> = ({ command }) => (
  <Box>
    <Spinner type="arrow3" />
    <Box paddingLeft={1}>Running command {command}...</Box>
  </Box>
);

const Results: React.FC<{
  code: number;
  out: string;
  err: Nullable<string>;
}> = ({ code, out, err }) => {
  useExit(code !== 0 ? new Error(code + '') : undefined);

  return (
    <Box>
      <Box>{out.trim()}</Box>
      {Nullable.maybe(
        null,
        err => (
          <Color red>{err}</Color>
        ),
        err
      )}
    </Box>
  );
};

const isComplete = (state: State): state is CompleteState =>
  Nullable.isSome(state.code);

const View: React.FC<State> = props => {
  if (Nullable.isNone(props.rc)) {
    return <Color red>RC file not loaded. Cannot build.</Color>;
  }

  if (props.rc instanceof Error) {
    return <Color red>RC file invalid. {props.rc.message}</Color>;
  }

  if (!isComplete(props)) {
    if (Nullable.isNone(props.command)) {
      return null;
    }
    return <Running command={props.command.replace(props.cwd + '/', '')} />;
  }

  return <Results code={props.code} out={props.out} err={props.err} />;
};

export default View;
