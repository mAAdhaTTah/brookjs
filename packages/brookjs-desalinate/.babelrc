{
  "plugins": [
    "syntax-flow",
    "transform-flow-strip-types"
  ],
  "env": {
    "test:node": {
      "presets": [
        ["brookjs", { "loose": false, "modules": "commonjs" }]
      ]
    },
    "test:browser": {
      "presets": [
        ["brookjs", { "loose": false, "modules": false }]
      ]
    },
    "commonjs": {
      "presets": [
        ["brookjs", { "loose": false, "modules": "commonjs" }]
      ]
    },
    "es": {
      "presets": [
        ["brookjs", { "loose": false, "modules": false }]
      ]
    }
  }
}
