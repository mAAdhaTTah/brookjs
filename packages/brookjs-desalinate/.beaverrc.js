export const dir = 'src';

/**
 * Mocha testing configuration.
 */
export const mocha = {
  reporter: 'spec',
  ui: 'bdd',
  requires: [
    '@babel/register',
    'react-testing-library',
    'jsdom-global/register',
    'raf/polyfill'
  ]
};

/**
 * Webpack build configuration.
 */
export const webpack = {
  entry: {
    app: 'app.js'
  },
  output: {
    path: 'dist/',
    filename: '[name].js'
  },
  modifier: x => x
};
