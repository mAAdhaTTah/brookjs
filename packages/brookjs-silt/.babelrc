{
  "env": {
    "test": {
      "presets": [
        ["brookjs", { "loose": false, "modules": "commonjs", "pragma": "h" }]
      ]
    },
    "commonjs": {
      "presets": [
        ["brookjs", { "loose": false, "modules": "commonjs", "pragma": "h" }]
      ]
    },
    "es": {
      "presets": [
        ["brookjs", { "loose": false, "modules": false, "pragma": "h" }]
      ]
    }
  }
}
