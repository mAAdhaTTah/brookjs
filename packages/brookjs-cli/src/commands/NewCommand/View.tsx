import React, { useContext, useEffect } from 'react';
import { Color, StdinContext, Box, Static } from 'ink';
import { toJunction } from 'brookjs-silt';
import TextInput from 'ink-text-input';
import SelectInput from 'ink-select-input';
import Spinner from 'ink-spinner';
import { Observable } from 'kefir';
import { Nullable } from 'typescript-nullable';
import { useExit } from '../../cli';
import {
  State,
  Configurable,
  unreachable,
  ConfiguredState,
  ConfiguringState,
  Log
} from './types';

type Props = State & {
  onChange: (value: string) => void;
  onSubmit: () => void;
  onConfirm: (value: boolean) => void;
};

const useOnSubmit = (onSubmit: () => void) => {
  const { stdin, setRawMode } = useContext(StdinContext);

  useEffect(() => {
    const onData = (data: Buffer) => {
      const s = data.toString();

      if (s === '\r') {
        onSubmit();
      }
    };

    setRawMode!(true);
    stdin.on('data', onData);

    return () => {
      // @TODO(mAAdhaTTah) This fails in node-pty â€“ not sure why
      // stdin.off is not a function
      stdin.off && stdin.off('data', onData);
      setRawMode!(false);
    };
  }, [stdin, setRawMode, onSubmit]);
};

type Question = {
  text: string;
  placeholder: string;
};

const questions: { [key in Configurable]: Question } = {
  version: {
    text: 'What is the application version?',
    placeholder: '0.0.0'
  },
  description: {
    text: 'What is the application description?',
    placeholder: 'A brookjs application'
  },
  dir: {
    text: 'Relative to the project, where will the source code live?',
    placeholder: 'src'
  },
  license: {
    text: 'Choose a license',
    placeholder: 'MIT'
  }
};

const ConfigureStep: React.FC<
  ConfiguringState & { onChange: (value: string) => void; onSubmit: () => void }
> = ({ configuring, config, onChange, onSubmit }) => {
  useOnSubmit(onSubmit);

  return (
    <Box flexDirection="column">
      <Color green>App name: {config.name}</Color>
      <Box>
        {questions[configuring].text}:{' '}
        <TextInput
          value={Nullable.withDefault('', config[configuring])}
          placeholder={questions[configuring].placeholder}
          focus={true}
          onChange={onChange}
        />
      </Box>
    </Box>
  );
};

const ConfirmStep: React.FC<
  ConfiguredState & { onConfirm: (value: boolean) => void }
> = ({ config: { name, version, description, dir, license }, onConfirm }) => (
  <Box flexDirection="column">
    <Box>Configuration:</Box>
    <Box>
      name: <Color green>{name}</Color>
    </Box>
    <Box>
      version: <Color green>{version}</Color>
    </Box>
    <Box>
      description: <Color green>{description}</Color>
    </Box>
    <Box>
      dir: <Color green>{dir}</Color>
    </Box>
    <Box>
      license: <Color green>{license}</Color>
    </Box>
    <Box>
      <SelectInput
        items={[
          {
            label: 'Yes',
            value: true
          },
          {
            label: 'No',
            value: false
          }
        ]}
        onSelect={({ value }) => onConfirm(value)}
      />
    </Box>
  </Box>
);

const CancelledStep: React.FC<{}> = () => {
  useExit();

  return <Box>Cancelled!</Box>;
};

const LogDisplay: React.FC<{ logs: Log[] }> = ({ logs }) => (
  <Box paddingBottom={2}>
    <Static>
      {logs.map((log, i) => (
        <Color key={i} green={log.level === 'ok'}>
          {log.msg}
        </Color>
      ))}
    </Static>
  </Box>
);

const CreatingStep: React.FC<{ name: string; logs: Log[] }> = ({
  name,
  logs
}) => (
  <Box flexDirection="column">
    <LogDisplay logs={logs} />
    <Box flexDirection="row">
      <Color green>
        <Spinner type="arrow3" />
      </Color>
      <Box marginLeft={1}>Creating app {name}</Box>
    </Box>
  </Box>
);

const ErrorStep: React.FC<{ error: Error }> = ({ error }) => {
  useExit(error);

  return (
    <Box flexDirection="column" paddingTop={2}>
      <Box>
        <Color red>Error running new</Color>
      </Box>
      <Box>
        <Color red>Message:</Color> {error.message}
      </Box>
    </Box>
  );
};

const CompleteStep: React.FC<{ name: string }> = ({ name }) => {
  useExit();

  return (
    <Box flexDirection="column" paddingTop={2}>
      <Box>
        <Color green>New brookjs application created at {name}</Color>
      </Box>
      <Box>
        Run <Color blue>npm i</Color> or <Color blue>yarn</Color> to bootstrap
      </Box>
    </Box>
  );
};

const View: React.FC<Props> = props => {
  if (Nullable.isNone(props.config.name)) {
    return (
      <Color red>
        Called <Color yellow>beaver new</Color> with no name.
      </Color>
    );
  }

  switch (props.step) {
    case 'configure':
      return <ConfigureStep {...props} />;
    case 'confirm':
      return <ConfirmStep {...props} />;
    case 'cancelled':
      return <CancelledStep />;
    case 'creating':
      return <CreatingStep name={props.config.name} logs={props.logs} />;
    case 'error':
      return <ErrorStep error={props.error} />;
    case 'complete':
      return <CompleteStep name={props.config.name} />;
    default:
      return unreachable(props);
  }
};

const events = {
  onChange: (e$: Observable<string, Error>) =>
    e$.map((value: string) => ({ type: 'INPUT', payload: { value } })),
  onSubmit: (e$: Observable<void, Error>) => e$.map(() => ({ type: 'SUBMIT' })),
  onConfirm: (e$: Observable<boolean, Error>) =>
    e$.map((value: boolean) => ({ type: 'CONFIRM', payload: { value } }))
};

export default toJunction<Props, typeof events>(events)(View);
