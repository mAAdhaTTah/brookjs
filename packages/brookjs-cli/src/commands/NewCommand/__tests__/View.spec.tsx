import React from 'react';
import { render } from 'ink-testing-library';
import { AppContext } from 'ink';
import sinon from 'sinon';
import { expect, use } from 'chai';
import { RootJunction } from 'brookjs-silt';
import { chaiPlugin } from 'brookjs-desalinate';
import Kefir from 'kefir';
import chaiJestSnapshot from 'chai-jest-snapshot';
import View from '../View';

const { plugin, value } = chaiPlugin({ Kefir });
use(plugin);
use(chaiJestSnapshot);

describe('NewCommand#View', () => {
  before(function() {
    chaiJestSnapshot.resetSnapshotRegistry();
  });

  beforeEach(function() {
    chaiJestSnapshot.configureUsingMochaContext(this);
  });

  describe('configure step', () => {
    it('should show an error when no name provided', () => {
      const { lastFrame, unmount } = render(
        <View
          error={null}
          logs={[]}
          result={null}
          step="configure"
          configuring="version"
          config={{
            name: null,
            version: null,
            description: null,
            dir: null,
            license: null
          }}
        />
      );

      expect(lastFrame()).to.matchSnapshot();

      unmount();
    });

    it('should show the first question', () => {
      const { lastFrame, unmount } = render(
        <View
          error={null}
          logs={[]}
          result={null}
          step="configure"
          configuring="version"
          config={{
            name: 'test-app',
            version: null,
            description: null,
            dir: null,
            license: null
          }}
        />
      );

      expect(lastFrame()).to.matchSnapshot();

      unmount();
    });

    it('should show the first question with version provided', () => {
      const { lastFrame, unmount } = render(
        <View
          error={null}
          logs={[]}
          result={null}
          step="configure"
          configuring="version"
          config={{
            name: 'test-app',
            version: '1.0.0',
            description: null,
            dir: null,
            license: null
          }}
        />
      );

      expect(lastFrame()).to.matchSnapshot();

      unmount();
    });

    it('should show the second question', () => {
      const { lastFrame, unmount } = render(
        <View
          error={null}
          logs={[]}
          result={null}
          step="configure"
          configuring="description"
          config={{
            name: 'test-app',
            version: '1.0.0',
            description: null,
            dir: null,
            license: null
          }}
        />
      );

      expect(lastFrame()).to.matchSnapshot();

      unmount();
    });

    it('should show the second question with description provided', () => {
      const { lastFrame, unmount } = render(
        <View
          error={null}
          logs={[]}
          result={null}
          step="configure"
          configuring="description"
          config={{
            name: 'test-app',
            version: '1.0.0',
            description: 'A test application',
            dir: null,
            license: null
          }}
        />
      );

      expect(lastFrame()).to.matchSnapshot();

      unmount();
    });

    it('should show the third question', () => {
      const { lastFrame, unmount } = render(
        <View
          error={null}
          logs={[]}
          result={null}
          step="configure"
          configuring="dir"
          config={{
            name: 'test-app',
            version: '1.0.0',
            description: 'A test application',
            dir: null,
            license: null
          }}
        />
      );

      expect(lastFrame()).to.matchSnapshot();

      unmount();
    });

    it('should show the third question with dir provided', () => {
      const { lastFrame, unmount } = render(
        <View
          error={null}
          logs={[]}
          result={null}
          step="configure"
          configuring="dir"
          config={{
            name: 'test-app',
            version: '1.0.0',
            description: 'A test application',
            dir: 'client',
            license: null
          }}
        />
      );

      expect(lastFrame()).to.matchSnapshot();

      unmount();
    });

    it('should show the fourth question', () => {
      const { lastFrame, unmount } = render(
        <View
          error={null}
          logs={[]}
          result={null}
          step="configure"
          configuring="license"
          config={{
            name: 'test-app',
            version: '1.0.0',
            description: 'A test application',
            dir: 'client',
            license: null
          }}
        />
      );

      expect(lastFrame()).to.matchSnapshot();

      unmount();
    });

    it('should show the fourth question with license provided', () => {
      const { lastFrame, unmount } = render(
        <View
          error={null}
          logs={[]}
          result={null}
          step="configure"
          configuring="license"
          config={{
            name: 'test-app',
            version: '1.0.0',
            description: 'A test application',
            dir: 'client',
            license: 'ISC'
          }}
        />
      );

      expect(lastFrame()).to.matchSnapshot();

      unmount();
    });

    it('should respond to input', () => {
      let r$;
      const { stdin } = render(
        <RootJunction
          root$={(root$: any) => {
            r$ = root$;
          }}
        >
          <View
            error={null}
            logs={[]}
            result={null}
            step="configure"
            configuring="version"
            config={{
              name: 'test-app',
              version: null,
              description: null,
              dir: null,
              license: null
            }}
          />
        </RootJunction>
      );

      expect(r$).to.emit(
        [
          value({ type: 'INPUT', payload: { value: 'test word' } }),
          value({ type: 'SUBMIT' })
        ],
        () => {
          stdin.write('test word');
          stdin.write('\r');
        }
      );
    });
  });

  describe('confirm step', () => {
    it('should display the current configruation', () => {
      const { lastFrame, unmount } = render(
        <View
          error={null}
          logs={[]}
          result={null}
          step="confirm"
          configuring={null}
          config={{
            name: 'test-app',
            version: '1.0.0',
            description: 'A test application',
            dir: 'client',
            license: 'ISC'
          }}
        />
      );

      expect(lastFrame()).to.matchSnapshot();

      unmount();
    });

    it('should respond to confirm', () => {
      let r$;
      const { stdin } = render(
        <RootJunction
          root$={(root$: any) => {
            r$ = root$;
          }}
        >
          <View
            error={null}
            logs={[]}
            result={null}
            step="confirm"
            configuring={null}
            config={{
              name: 'test-app',
              version: '1.0.0',
              description: 'A test application',
              dir: 'client',
              license: 'ISC'
            }}
          />
        </RootJunction>
      );

      expect(r$).to.emit(
        [value({ type: 'CONFIRM', payload: { value: true } })],
        () => {
          stdin.write('\r');
        }
      );
    });

    it('should respond to confirm with false if selected', () => {
      let r$;
      const { stdin } = render(
        <RootJunction
          root$={(root$: any) => {
            r$ = root$;
          }}
        >
          <View
            error={null}
            logs={[]}
            result={null}
            step="confirm"
            configuring={null}
            config={{
              name: 'test-app',
              version: '1.0.0',
              description: 'A test application',
              dir: 'client',
              license: 'ISC'
            }}
          />
        </RootJunction>
      );

      expect(r$).to.emit(
        [value({ type: 'CONFIRM', payload: { value: false } })],
        () => {
          stdin.write('\u001B[B');
          stdin.write('\r');
        }
      );
    });
  });

  describe('cancelled', () => {
    it('should render cancelled message and call exit', done => {
      const exit = sinon.spy();
      const { lastFrame, unmount } = render(
        <AppContext.Provider value={{ exit }}>
          <View
            error={null}
            logs={[]}
            result={null}
            step="cancelled"
            configuring={null}
            config={{
              name: 'test-app',
              version: '1.0.0',
              description: 'A test application',
              dir: 'client',
              license: 'ISC'
            }}
          />
        </AppContext.Provider>
      );

      setTimeout(() => {
        expect(lastFrame()).to.matchSnapshot();
        expect(exit.calledOnce).to.equal(true);

        unmount();

        done();
      }, 10);
    });
  });

  describe('creating', () => {
    it('should render spinner', () => {
      const { lastFrame, unmount } = render(
        <View
          error={null}
          logs={[]}
          result={null}
          step="creating"
          configuring={null}
          config={{
            name: 'test-app',
            version: '1.0.0',
            description: 'A test application',
            dir: 'client',
            license: 'ISC'
          }}
        />
      );

      expect(lastFrame()).to.matchSnapshot();

      unmount();
    });

    it('should render all logs', () => {
      const { lastFrame, unmount } = render(
        <View
          error={null}
          logs={[
            {
              level: 'notice',
              msg: 'This is a notice mesasge.'
            },
            {
              level: 'warn',
              msg: 'This is a warn message.'
            },
            {
              level: 'error',
              msg: 'This is a error message.'
            },
            {
              level: 'ok',
              msg: 'This is a ok message.'
            }
          ]}
          result={null}
          step="creating"
          configuring={null}
          config={{
            name: 'test-app',
            version: '1.0.0',
            description: 'A test application',
            dir: 'client',
            license: 'ISC'
          }}
        />
      );

      expect(lastFrame()).to.matchSnapshot();

      unmount();
    });
  });

  describe('error', () => {
    it('should render error message and call exit', done => {
      const exit = sinon.spy();
      const error = new Error('Creating failed');
      const { lastFrame, unmount } = render(
        <AppContext.Provider value={{ exit }}>
          <View
            error={error}
            logs={[]}
            result={null}
            step="error"
            configuring={null}
            config={{
              name: 'test-app',
              version: '1.0.0',
              description: 'A test application',
              dir: 'client',
              license: 'ISC'
            }}
          />
        </AppContext.Provider>
      );

      setTimeout(() => {
        expect(lastFrame()).to.matchSnapshot();
        expect(exit.calledOnce).to.equal(true);
        expect(exit.args[0][0]).to.equal(error);

        unmount();

        done();
      }, 10);
    });
  });

  describe('complete', () => {
    it('should render complete message and call exit', done => {
      const exit = sinon.spy();
      const { lastFrame, unmount } = render(
        <AppContext.Provider value={{ exit }}>
          <View
            error={null}
            logs={[]}
            result={{ success: true, time: Date.now(), actions: [] }}
            step="complete"
            configuring={null}
            config={{
              name: 'test-app',
              version: '1.0.0',
              description: 'A test application',
              dir: 'client',
              license: 'ISC'
            }}
          />
        </AppContext.Provider>
      );

      setTimeout(() => {
        expect(lastFrame()).to.matchSnapshot();
        expect(exit.calledOnce).to.equal(true);

        unmount();

        done();
      }, 10);
    });
  });
});
