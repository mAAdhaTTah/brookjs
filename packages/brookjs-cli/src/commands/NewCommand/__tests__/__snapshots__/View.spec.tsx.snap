// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCommand#View cancelled should render cancelled message and call exit 1`] = `"Cancelled!"`;

exports[`NewCommand#View complete should render complete message and call exit 1`] = `
"

[32mNew brookjs application created at test-app[39m
Run [34mnpm i[39m or [34myarn[39m to bootstrap"
`;

exports[`NewCommand#View configure step should show an error when no name provided 1`] = `"[31mCalled [33mbeaver new[31m with no name.[39m"`;

exports[`NewCommand#View configure step should show an error when no name provided 2`] = `"[31mCalled [33mbeaver new[31m with no name.[39m"`;

exports[`NewCommand#View configure step should show the first question 1`] = `
"[32mApp name: test-app[39m
What is the application version?: [2m0.0.0[22m"
`;

exports[`NewCommand#View configure step should show the first question with version provided 1`] = `
"[32mApp name: test-app[39m
What is the application version?: 1.0.0[7m [27m"
`;

exports[`NewCommand#View configure step should show the fourth question 1`] = `
"[32mApp name: test-app[39m
Choose a license: [2mMIT[22m"
`;

exports[`NewCommand#View configure step should show the fourth question with license provided 1`] = `
"[32mApp name: test-app[39m
Choose a license: ISC[7m [27m"
`;

exports[`NewCommand#View configure step should show the second question 1`] = `
"[32mApp name: test-app[39m
What is the application description?: [2mA brookjs application[22m"
`;

exports[`NewCommand#View configure step should show the second question with description provided 1`] = `
"[32mApp name: test-app[39m
What is the application description?: A test application[7m [27m"
`;

exports[`NewCommand#View configure step should show the third question 1`] = `
"[32mApp name: test-app[39m
Relative to the project, where will the source code live?: [2msrc[22m"
`;

exports[`NewCommand#View configure step should show the third question with dir provided 1`] = `
"[32mApp name: test-app[39m
Relative to the project, where will the source code live?: client[7m [27m"
`;

exports[`NewCommand#View confirm step should display the current configruation 1`] = `
"Configuration:
name: [32mtest-app[39m
version: [32m1.0.0[39m
description: [32mA test application[39m
dir: [32mclient[39m
license: [32mISC[39m
[34m‚ùØ[39m [39m[34mYes[39m[39m
  No"
`;

exports[`NewCommand#View creating should render all logs 1`] = `
"This is a notice mesasge.
This is a warn message.
This is a error message.
[32mThis is a ok message.[39m


[32m‚ñπ‚ñπ‚ñπ‚ñπ‚ñπ[39m [39mCreating app test-app[39m"
`;

exports[`NewCommand#View creating should render spinner 1`] = `
"

[32m‚ñπ‚ñπ‚ñπ‚ñπ‚ñπ[39m [39mCreating app test-app[39m"
`;

exports[`NewCommand#View error should render error message and call exit 1`] = `
"

[31mError running new[39m
[31mMessage:[39m Creating failed"
`;
